src/detection_win.cpp

```diff
@@ -423,6 +424,21 @@ void ExtractDeviceInfo(HDEVINFO hDevInfo, SP_DEVINFO_DATA* pspDevInfoData, TCHAR
 		// Use this to extract VID / PID
 		extractVidPid(buf, resultItem);
 	}
+
+	DWORD dwCap = 0x0;
+	if (DllSetupDiGetDeviceRegistryProperty(hDevInfo, pspDevInfoData, SPDRP_CAPABILITIES, &DataT, (PBYTE)&dwCap, 4, &nSize)) {
+		// If device has a CM_DEVCAP_UNIQUEID capability, then we assume it has a serial number
+		if ((dwCap & CM_DEVCAP_UNIQUEID) == CM_DEVCAP_UNIQUEID) {
+			if (DllSetupDiGetDeviceInstanceId(hDevInfo, pspDevInfoData, buf, buffSize, &nSize)) {
+				// extract Serial Number
```

Comment:
Let's add a comment about what we are trying to parse

```
// Format: <device-ID>\<instance-specific-ID>
//
// Ex. `USB\VID_2109&PID_8110\5&376ABA2D&0&21`
//  - `<device-ID>`: `USB\VID_2109&PID_8110`
//  - `<instance-specific-ID>`: `5&376ABA2D&0&21`
//
// [Device instance IDs](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-instance-ids) ->
//  - [Device IDs](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-ids) -> [Hardware IDs](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/hardware-ids) -> [Device identifier formats](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/device-identifier-formats) -> [Identifiers for USB devices](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/identifiers-for-usb-devices)
//     - [Standard USB Identifiers](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/standard-usb-identifiers)
//     - [Special USB Identifiers](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/special-usb-identifiers)
//  - [Instance specific ID](https://docs.microsoft.com/en-us/windows-hardware/drivers/install/instance-ids) 
```

---

From https://github.com/MadLittleMods/node-usb-detection/pull/50#discussion_r153106778

Reply:
I am OK with that